### PR TITLE
Improve cargo-sweep.sh

### PR DESCRIPTION
**Pull request by pzmarzly**
See conversation in #212. cargo-sweep will no longer print "error" when cargo-sweep is already installed. Also, updating it should now work.

### CODE REVIEW

**Comment by realcr** [comment]
@pzmarzly : Thanks, I will merge this. I noticed some strange error in one of the integration tests in macOS, I try to restart it. I will check it later and fix if necessary in another PR.

**Comment by pzmarzly** [comment]
I also experienced the same issue in #215 after `Apply cargo fmt` commit, restarting the macOS job helped.

**Comment by realcr** [comment]
> I also experienced the same issue in #215 after Apply cargo fmt commit, restarting the macOS job helped.

I added some code to document what happens in the bad integration test (On MacOS). Next time it happens could you include the error? I'm very interested to see what happens there. My guess is some kind of networking problem.

**Comment by realcr** [comment]
@pzmarzly : I think that I fixed the macOS bug, but I accidentally pushed the fix directly into master. Sorry about that \:

**Comment by pzmarzly** [comment]
Ready for merging

**Comment by realcr** [comment]
@pzmarzly : Thanks! Could you check my question above about duplicate `if` clause in the bash script? Is there a way we can make it more compact?

File: travis/trusty/post/cargo-sweep.sh [hunk 1]
```diff
@@ -1,9 +1,18 @@
 #!/usr/bin/env bash
 
-# Install cargo-sweep. If installed, cargo failure will be ignored.
-# TODO: parse ~/.cargo/.crates.toml to determine version
-#       maybe use cargo-update library for that
-cargo install cargo-sweep --vers 0.4.1 || true
+CARGO_SWEEP_VERSION=0.4.1
+
+# Install cargo-sweep if absent
+if [ ! -f ~/.cargo/bin/cargo-sweep ]; then
```
[the remark this answers is not in the document]
Comment (pzmarzly) [reply travis/trusty/post/cargo-sweep.sh]: As long as no one puts `set -o pipefail` (which would make `grep` exit code relevant), this `if` can indeed be removed. I guess I can do that now.

**Comment by realcr** [comment]
@pzmarzly : Thanks!

**Pull request merged**